### PR TITLE
Subprocess changes and dedent

### PR DESCRIPTION
**Pull request by chrisglass**
This branch provides two nitpicky non-functional changes:

1. Switch from using os.system() to the higher-level subprocess.call() (the recommended interface for shelling out in python). In particular this allows to retrieve non-zero exit codes instead of getting the syscall + exit code integer from os.system.
2. Use textwrap.dedent() to have the yaml snippets aligh with the rest of the code instead of "leaking" to the left-hand side of the text file.

These changes are entirely "cosmetic" and should not affect the functionality at all.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.

### CODE REVIEW

**Review by simu — APPROVED** [review]
LGTM

File: commodore/component/compile.py [hunk 1]
```diff
@@ -42,51 +43,48 @@ def compile_component(config: Config, component_path, value_files, search_paths,
...
-  App: ArgoApp,
-  Project: ArgoProject,
-}
-""")
+            file.write(dedent("""
+                local ArgoApp(component, namespace, project='', secrets=true) = {};
+                local ArgoProject(name) = {};
+
+                {
+                App: ArgoApp,
```
Comment (srueg) [inline commodore/component/compile.py:85]:
```suggestion
                  App: ArgoApp,
```

File: commodore/component/compile.py [hunk 2]
```diff
@@ -42,51 +43,48 @@ def compile_component(config: Config, component_path, value_files, search_paths,
...
+                Project: ArgoProject,
```
Comment (srueg) [inline commodore/component/compile.py:86]:
```suggestion
                  Project: ArgoProject,
```

File: tests/test_component_compile.py [hunk 3]
```diff
@@ -14,15 +18,14 @@ def _prepare_component(tmp_path, component_name='test-component'):
...
-{
-  "test_service_account": kube.ServiceAccount('test') {
-    metadata+: {
-      namespace: params.namespace,
-    },
-  },
-}
-''')
+        file.write(dedent('''
+            {
```
Comment (srueg) [inline tests/test_component_compile.py:22]: Indentation broke a little here.

**Review by srueg — CHANGES_REQUESTED** [review]

File: commodore/component/compile.py (hunk 1 above)
Comment (chrisglass) [reply commodore/component/compile.py:85]: Oh, good catch! It's valid JSONnet, so it did escape the unit tests. Will change (and other places too of course).

File: tests/test_component_compile.py (hunk 3 above)
Comment (chrisglass) [reply tests/test_component_compile.py:22]: Yep, same problem as above. I'll fix it.

Good catch!

Comment (chrisglass) [reply tests/test_component_compile.py:22]: Done.

**Review by srueg — APPROVED** [review]

**Pull request merged**